sympy/core/expr.py

```diff
@@ -826,7 +827,12 @@ def _eval_interval(self, x, a, b):
         value = B - A
 
         if a.is_comparable and b.is_comparable:
-            singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
+            if a < b:
+                domain = Interval(a, b)
+            else:
+                domain = Interval(b, a)
+            singularities = solveset(self.cancel().as_numer_denom()[1], x)
+            singularities = list(Intersection(domain, singularities))
```

Comment:
Can we use `domain = domain` as a kwarg for `solveset` function instead of taking an explicit intersection?